### PR TITLE
qa: full set of roles in health-openattic test, flexible number of MONs, and other gems

### PR DESCRIPTION
**Pull request by smithfarm**
Improvements to the qa suite in this PR:

* health-openattic.sh now installs a full cluster with rgw, mds, igw, and ganesha roles (rgw and openattic nodes are colocated only in single-node configurations)
* number of MONs is now flexible - either one or three, depending on how many nodes are available
* separate policy_cfg_storage from policy_cfg_base -> the former now takes options specifying number of non-storage nodes ("clients") and whether or not OSDs are encrypted
* rename validate_rados_put function to rados_write_test and use it in both health-ok-encrypted.sh and health-ok.sh
* implement an --ssl option to health-rgw.sh (and drop health-rgw-ssl.sh as a separate file)

### CODE REVIEW

**Comment by smithfarm** [comment]
test passes on three nodes: http://149.202.189.253:8081/ubuntu-2017-09-15_17:38:21-deepsea:basic:openattic-ses5---basic-openstack/

**Comment by smithfarm** [comment]
full run **running** http://149.202.189.253:8081/ubuntu-2017-09-15_20:44:30-deepsea:basic-ses5---basic-openstack/

**Comment by smithfarm** [comment]
This is still missing:

* NFS-Ganesha demo users
* mds role

**Comment by smithfarm** [comment]
health-openattic
**running** http://149.202.189.253:8081/ubuntu-2017-09-15_22:09:48-deepsea:basic:openattic-ses5---basic-openstack/

**Comment by smithfarm** [comment]
Also tested in AWS and it really does deploy openATTIC - I logged in!

File: qa/common/common.sh [hunk 1]
```diff
@@ -172,38 +164,78 @@ config/stack/default/ceph/cluster.yml
...
+  # second argument controls whether OSDs are encrypted; default not encrypted
+  local CLIENTS=$1
+  test -z "$CLIENTS" && CLIENTS=0
+  local ENCRYPTION=$2
 
-function policy_cfg_client {
-  cat <<EOF >> /srv/pillar/ceph/proposals/policy.cfg
+  local PROFILE="default"
+  if [ -n "$ENCRYPTION" ] ; then
+     PROFILE=dmcrypt
```
Comment (jschmid1) [inline qa/common/common.sh]: consistency nit:

``` bash
PROFILE="dmcrypt"
```

Comment (smithfarm) [reply qa/common/common.sh]: @jschmid1 Thanks, fixed.

**Comment by smithfarm** [comment]
Full run is here: http://149.202.189.253:8081/ubuntu-2017-09-18_15:02:02-deepsea:basic-ses5---basic-openstack/

There was one failure, which is fixed by 2c29ccb

**Pull request merged**